chrome/content/zotero/xpcom/server.js

```diff
@@ -384,21 +384,34 @@ Zotero.Server.DataListener.prototype._processEndpoint = function(method, postDat
 		if(postData && this.contentType) {
 			// check that endpoint supports contentType
 			var supportedDataTypes = endpoint.supportedDataTypes;
-			if(supportedDataTypes && supportedDataTypes.indexOf(this.contentType) === -1) {
+			if(supportedDataTypes && 
+				(supportedDataTypes == '*' || supportedDataTypes.indexOf(this.contentType) === -1)) {
```

Comment:
Wouldn't this make it so that including a `Content-Type` header for any `supportedDataTypes: '*'` endpoint would result in a 400?